repository_service_tuf/cli/admin/metadata.py

```diff
@@ -0,0 +1,596 @@
+    )
+    while True:
+        signing_list = [
+            key.to_dict() for key in current_root.signing_keys.values()
+        ]
+        keys_table = _create_keys_table(signing_list, current_root, False)
+        console.print("\nHere are the current root signing keys:")
+        console.print(keys_table)
+        keys_amount = len(current_root.signing_keys)
+        if root_threshold <= keys_amount:
+            agree = prompt.Confirm.ask("\nDo you want to add a new key?")
+            if not agree:
+                return
+        else:
+            remaining = root_threshold - keys_amount
+            console.print(f"You must add {remaining} more signing key(s)")
+
+        root_key: RSTUFKey = _get_key(Root.type)
+        if root_key.error:
+            console.print(f":cross_mark: [red]Failed[/]: {root_key.error}")
+            continue
+
+        if root_key.key["keyid"] == current_root.online_key["keyid"]:
+            console.print(
+                ":cross_mark: [red]Failed[/]: This is the current online key. "
+                "Cannot be added"
+            )
+            continue
+
+        if current_root.is_keyid_used(root_key.key["keyid"]):
+            console.print(":cross_mark: [red]Failed[/]: Key is already used")
+            continue
+
+        root_key.name = prompt.Prompt.ask(
+            "[Optional] Give a [green]name/tag[/] to the key"
+        )
+
+        current_root.add_key(root_key)
+
+
+def _get_positive_int_input(msg: str, input_name: str, default: Any) -> int:
+    input: int = 0
+    while True:
+        input = prompt.IntPrompt.ask(msg, default=default, show_default=True)
+        if input >= 1:
+            return input
+
+        console.print(f"{input_name} must be at least 1")
+
+
+def _modify_expiration(current_root: RootInfo):
+    console.print(markdown.Markdown(EXPIRY_CHANGES_MSG), width=100)
+    console.print("\n")
+    change: bool
+    while True:
+        console.print(
+            f"Current root expiration: [cyan]{current_root.expiration_str}[/]",
+            highlight=False,  # disable built-in rich highlight
+        )
+        if current_root.expiration < (datetime.now() + timedelta(days=1)):
+            console.print("Root root has expired - expiration must be extend")
+            change = True
+
+        else:
+            change = prompt.Confirm.ask(
+                "Do you want to extend the [cyan]root's expiration[/]?"
+            )
+
+        if not change:
+            console.print("Skipping root expiration changes")
+            return
+        else:
+            m = "Days to extend [cyan]root's expiration[/] starting from today"
+            bump = _get_positive_int_input(m, "Expiration extension", 365)
+            new_expiry = datetime.now() + timedelta(days=bump)
+            new_exp_str = new_expiry.strftime("%Y-%b-%d")
+            agree = prompt.Confirm.ask(
+                f"New root expiration: [cyan]{new_exp_str}[/]. Do you agree?"
```

Comment:
Probably a matter of taste, but I think it's nicer to re-use the prompts from above as I suggested. The flow is a bit awkward if the user does not agree here. It will look like.
```
...
New root expiration: <DATE>. Do you agree?
Current root expiration: <DATE>
Do you want to extend?
...
```

Reply:
> Probably a matter of taste, but I think it's nicer to re-use the prompts from above[ as I suggested](https://github.com/repository-service-tuf/repository-service-tuf-cli/pull/284#discussion_r1233898921). The flow is a bit awkward if the user does not agree here. It will look like.
> 
> ```
> ...
> New root expiration: <DATE>. Do you agree?
> Current root expiration: <DATE>
> Do you want to extend?
> ...
> ```

Reply:
I am not sure I understand your idea here, but can it wait for another pr?
Feel free to suggest a pr on top of mine.